src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+public class GatherSplitReadsSpark extends GATKSparkTool
+{
+    private static final long serialVersionUID = 1L;
+    private static final int KLEN = 63;
+
+    @Argument(doc = "dir for output FASTQs", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    private String outputDir;
+
+    // This is a path to a file of kmers that appear too frequently in the reference to be usable as probes to localize
+    // reads.  We don't calculate it here, because it depends only on the reference.
+    // The program FindBadGenomicKmers can produce such a list for you.
+    @Argument(doc = "ubiquitous kmer list", shortName = "KS", fullName = "kmerKillSet", optional = false)
+    private String kmerKillSet;
+
+    @Override
+    public boolean requiresReads()
+    {
+        return true;
+    }
+
+    @Override
+    protected void runTool( final JavaSparkContext ctx )
+    {
+        final SAMFileHeader header = getHeaderForReads();
+        if ( header.getSortOrder() != SAMFileHeader.SortOrder.coordinate )
+            throw new GATKException("The BAM must be coordinate sorted.");
+
+        // Process the input data to produce a map of Kmer -> List of breakpoint IDs.
+        // (A breakpoint ID is just an arbitrary identifier for the putative breakpoint.)
+        // See the doc for findClusters to see a description of how this big Spark pipeline works.
+        final Broadcast<Map<Kmer,List<Long>>> broadcastMap = ctx.broadcast(findClusters(ctx));
+
+        // Process the input again, this time pulling all reads that contain kmers associated with a given breakpoint,
+        // and writing those reads into a separate FASTQ for each breakpoint.
+        getGoodReads()
+            .mapPartitionsToPair(readItr -> new ReadPuller(readItr,broadcastMap.value()))
+            .groupByKey()
+            .foreach(tupl -> writeFastq(tupl._1,tupl._2.iterator(),outputDir));
+    }
+
+    public JavaRDD<GATKRead> getGoodReads() {
+        return getUnfilteredReads().filter(read ->
+            !read.isSecondaryAlignment() && !read.isSupplementaryAlignment() &&
+                    !read.isDuplicate() && !read.failsVendorQualityCheck());
+    }
+
+    // Read a file of kmers to ignore.
+    // Each line must be exactly KLEN characters long, and must match [ACGT]*.
+    private Set<Kmer> readKillSet()
+    {
+        final Set<Kmer> killSet = new HashSet<>();
+
+        try
+        {
+            final BufferedReader rdr = new BufferedReader(new FileReader(kmerKillSet));
+            String line;
+            while ( (line = rdr.readLine()) != null )
+            {
+                if ( line.length() != KLEN )
+                    throw new GATKException("Kmer kill set contains a line of length " + line.length() +
+                            " but we were expecting K=" + KLEN);
+
+                final StringKmerizer kmerizer = new StringKmerizer(line, KLEN);
```

Comment:
Sorry, there was so little context I didn't realize where we were in the code.  The only reason not to use Kmerizer.toKmer here is that that method will generate a rather generic message for the exception (`"Can't make a Kmer from '"+seq+"'"`) rather than the one I'm generating 2 lines below (which makes it clear where the bogus sequence came from:  the kill file).  Easier for users to understand what we were doing when the bad thing happened.